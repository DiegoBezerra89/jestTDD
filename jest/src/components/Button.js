export function Button(props) {
  //SNAPSHOT
  // return <div></div>
  // return (
  //   <button>
  //   </button>
  // )
  //## GREEN TEST 01
  const { content, handleClick, classe } = props;
  return (
    //<button className={classe} onClick={handleClick}>
    // {content}
    //</button>
    <button className={classe} onClick={handleClick}>
      {content}
    </button>
  )
}