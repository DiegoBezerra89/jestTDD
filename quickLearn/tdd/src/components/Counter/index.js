import React, { useState } from "react";
import "./index.css";

/* O contador deve iniciar com o valor 0
function Counter() {
  return <div className="counter">
    <h1>0</h1>
  </div>
}*/

/* O título deve conter a classe counter__title
function Counter() {
  return <div className="counter">
    <h1 className="counter__title">0</h1>
  </div>
}*/

/* não deve iniciar o título com a classe counter__title--increment
function Counter() {
  return <div className="counter">
    <h1 className="counter__title counter__title--increment">0</h1>
  </div>
}*/


/* não deve iniciar o título com a classe counter__title--increment
function Counter() {
  return <div className="counter">
    <h1 className="counter__title counter__title--increment counter__title--decrement">0</h1>
  </div>
}*/

/*deve conter um botão incrementar 
function Counter() {
  return <div className="counter">
    <h1 className="counter__title">0</h1>
    <section className="buttons">
        <button className="button button--increment">
          incrementar
        </button>
        <button className="button button--decrement">
          decrementar
        </button>
      </section>
  </div>
}
*/

/*deve decrementar - 1 ao clicar no botão decrementar*/
function Counter() {
  const [ counter , setCounter] = useState(0);

  return <div className="counter">
    <h1 className="counter__title">{counter}</h1>
    <section className="buttons">
        <button className="button button--decrement" onClick={() => setCounter(counter - 1)}>
          decrementar
        </button>
        <button className="button button--increment" onClick={() =>  setCounter(counter + 1)}>
          incrementar
        </button>
      </section>
  </div>
}

export default Counter;
