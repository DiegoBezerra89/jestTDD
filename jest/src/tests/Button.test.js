import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { Button } from '../components/Button';

//## RED Test ##
it('render decrement correctly [Red Test]', () => {
  const tree = renderer.create(
    <Button content=' - Decrement '/>
  ).toJSON()
  expect(tree).toMatchSnapshot();
})

//## GREEN TEST 01 ##
it('render decrement correctly [Green test]', () => {
  const { root } = renderer.create(
    <Button content=' - Decrement '/>
  )
  expect(root.props.content).toBe(' - Decrement ');
})

it('render increment correctly [Red Test]', () => {
  const { root } = renderer.create(
    <Button content=' + Increment '/>
  )
  expect(root.props.content).toBe(' + Increment ');
})

it('render className correctly', () => {
  const { root } = renderer.create(
    <Button content=' + Increment ' classe='increment' />
  )
  expect(root.props.classe).toBe('increment')
})

//## Snapshot ##
it('renders decrement without errors', () => {
  const tree = renderer.create(
    <Button classe='decrement' content=' - Decrement ' />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders increment without errors', () => {
  const tree = renderer.create(
    <Button classe='increment' content=' + Increment ' />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})



