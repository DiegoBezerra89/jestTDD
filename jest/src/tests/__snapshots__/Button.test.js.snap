// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render decrement correctly [Red Test] 1`] = `
<button>
   - Decrement 
</button>
`;

exports[`renders decrement without errors 1`] = `
<button
  className="decrement"
>
   - Decrement 
</button>
`;

exports[`renders increment without errors 1`] = `
<button
  className="increment"
>
   + Increment 
</button>
`;
